tests/providers/slack/chat-message.ts

```diff
@@ -0,0 +1,195 @@
+      idPrefix: `${provider.getProviderId()}-${connection.profile.id}`,
+      timeOrderAttribute: "insertedAt",
+      batchSizeLimitAttribute: "batchSize",
+    };
+  });
+
+  // Test fetching data for Slack Chat
+  describe(`Fetch ${providerId} data`, () => {
+
+    it(`Can pass basic tests: ${handlerName}`, async () => {
+      // Build the necessary test objects
+      const { api, handler, provider } = await CommonTests.buildTestObjects(
+        providerId,
+        SlackChatMessageHandler,
+        providerConfig,
+        connection
+      );
+
+      // Set the handler configuration
+      handler.setConfig(handlerConfig);
+
+      try {
+        // Set up initial sync position
+        const syncPosition: SyncHandlerPosition = {
+          _id: `${providerId}-${handlerName}`,
+          providerId,
+          handlerId: handler.getId(),
+          accountId: provider.getAccountId(),
+          status: SyncHandlerStatus.ENABLED,
+        };
+
+        // Start the sync process
+        const response = await handler._sync(api, syncPosition);
+        const results = <SchemaRecord[]>response.results;
+
+        // Extract chat groups and messages from the results
+        const chatGroups = <SchemaSocialChatGroup[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_GROUP);
+        const chatMessages = <SchemaSocialChatMessage[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_MESSAGE);
+
+        // Ensure results are returned
+        assert.ok(results && results.length, "Have results returned");
+
+        // Check IDs in the returned items
+        CommonTests.checkItem(results[0], handler, provider);
+
+        // Verify sync status is active
+        assert.equal(
+          SyncHandlerStatus.SYNCING,
+          response.position.status,
+          "Sync is active"
+        );
+
+        // Ensure the message batch per group works
+        const firstGroupId = chatMessages[0].groupId;
+        const firstGroupMessages = chatMessages.filter(msg => msg.groupId === firstGroupId);
+        assert.equal(firstGroupMessages.length, handlerConfig.messagesPerGroupLimit, "Processed correct number of messages per group");
+
+         /**
+          * Start the second sync batch process
+          */
+        const secondBatchResponse = await handler._sync(api, response.position);
+        const secondBatchResults = <SchemaRecord[]>secondBatchResponse.results;
+
+        // Extract chat groups and messages from the second batch results
+        const secondBatchChatGroups = <SchemaSocialChatGroup[]>secondBatchResults.filter(result => result.schema === CONFIG.verida.schemas.CHAT_GROUP);
+        const secondBatchChatMessages = <SchemaSocialChatMessage[]>secondBatchResults.filter(result => result.schema === CONFIG.verida.schemas.CHAT_MESSAGE);
+
+        // Ensure second batch results are returned
+        assert.ok(secondBatchResults && secondBatchResults.length, "Have second batch results returned");
+
```

Comment:
Check second batch items aren't in the first batch
Check second batch first item is older than the first batch last item

Reply:
> Check second batch items aren't in the first batch 

Done
>Check second batch first item is older than the first batch last item

A batch has multiple channels, can't do cross-channel ordering